### PR TITLE
feat: add `gcp_redis_cluster` table and improve `gcp_redis_instance`

### PR DESCRIPTION
**Pull request by pdecat**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
Add passing integration test logs here
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  create_time,
  location,
  size_gb,
  precise_size_gb,
  psc_connections[0] ->> 'address' as address
from
  gcp_redis_cluster;

+----------------------------------+---------------------------+--------------+---------+-----------------+-----------+
| name                             | create_time               | location     | size_gb | precise_size_gb | address   |
+----------------------------------+---------------------------+--------------+---------+-----------------+-----------+
| test-pdecat-memory-redis-cluster | 2024-10-29T16:44:06+01:00 | europe-west9 | 2       | 1.4             | 10.0.0.15 |
+----------------------------------+---------------------------+--------------+---------+-----------------+-----------+

```

```
> select
  name,
  create_time,
  location,
  psc_connections[0] ->> 'address' as address
from
  gcp_redis_cluster
where
  authorization_mode != 1;
+----------------------------------+---------------------------+--------------+-----------+
| name                             | create_time               | location     | address   |
+----------------------------------+---------------------------+--------------+-----------+
| test-pdecat-memory-redis-cluster | 2024-10-29T16:44:06+01:00 | europe-west9 | 10.0.0.15 |
+----------------------------------+---------------------------+--------------+-----------+
```

```
> select
  name,
  create_time,
  location,
  psc_connections[0] ->> 'address' as address
from
  gcp_redis_cluster
where
  create_time >= current_timestamp - interval '7 days';
+----------------------------------+---------------------------+--------------+-----------+
| name                             | create_time               | location     | address   |
+----------------------------------+---------------------------+--------------+-----------+
| test-pdecat-memory-redis-cluster | 2024-10-29T16:44:06+01:00 | europe-west9 | 10.0.0.15 |
+----------------------------------+---------------------------+--------------+-----------+
```

```
> select
  name,
  create_time,
  location,
  node_type,
  size_gb,
  replica_count,
  shard_count
from
  gcp_redis_cluster
where
  name = 'test-pdecat-memory-redis-cluster'
  and location = 'europe-west9';

+----------------------------------+---------------------------+--------------+-----------+---------+---------------+-------------+
| name                             | create_time               | location     | node_type | size_gb | replica_count | shard_count |
+----------------------------------+---------------------------+--------------+-----------+---------+---------------+-------------+
| test-pdecat-memory-redis-cluster | 2024-10-29T16:44:06+01:00 | europe-west9 | 1         | 2       | 0             | 1           |
+----------------------------------+---------------------------+--------------+-----------+---------+---------------+-------------+
```

```
> select
  name,
  create_time,
  location,
  psc_connections[0] ->> 'address' as address
from
  gcp_redis_cluster
where
  transit_encryption_mode != 2;
+----------------------------------+---------------------------+--------------+-----------+
| name                             | create_time               | location     | address   |
+----------------------------------+---------------------------+--------------+-----------+
| test-pdecat-memory-redis-cluster | 2024-10-29T16:44:06+01:00 | europe-west9 | 10.0.0.15 |
+----------------------------------+---------------------------+--------------+-----------+
```
</details>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,40 +5,41 @@ go 1.22.4
...
 
 require (
-	cloud.google.com/go v0.112.1 // indirect
-	cloud.google.com/go/compute v1.24.0 // indirect
-	cloud.google.com/go/iam v1.1.7 // indirect
-	cloud.google.com/go/redis v1.14.2
-	cloud.google.com/go/storage v1.38.0 // indirect
+	cloud.google.com/go v0.115.0 // indirect
+	cloud.google.com/go/iam v1.1.12 // indirect
+	cloud.google.com/go/redis v1.16.5
```
Comment (pdecat) [inline go.mod:41]: Updating this package is required to get the following fields of Redis Clusters:
* `NodeType`
* `PersistenceConfig`
* `RedisConfigs`
* `PreciseSizeGb`
* `ZoneDistributionConfig`
* `DeletionProtectionEnabled`

https://github.com/googleapis/google-cloud-go/commit/d5150d34eabac0218cbd16a9bbdaaaf019cf237d#diff-95ecadd2969eaca0ed1ad11a0f58b28001694a2a8db14aef83934bada5f17e68R1039

File: gcp/table_gcp_redis_cluster.go [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+				Description: "Precise value of redis memory size in GB for the entire cluster.",
+				Type:        proto.ColumnType_DOUBLE,
+			},
+			{
+				Name:        "deletion_protection_enabled",
+				Description: "The delete operation will fail when the value is set to true.",
+				Type:        proto.ColumnType_JSON,
+			},
+			// FIXME: this is missing from the Go SDK
+			// https://github.com/googleapis/google-cloud-go/issues/11061
```
Comment (pdecat) [inline gcp/table_gcp_redis_cluster.go:136]: One of the maintainers already reported this issue internally.

**Comment by pdecat** [comment]
FWIW, I've planned to add Memorystore for Memcached next, and Memorystore for Valkey as soon as it is available in the Go SDK.

File: gcp/plugin.go [hunk 3]
```diff
@@ -36,6 +37,40 @@ func Plugin(ctx context.Context) *plugin.Plugin {
 		ConnectionConfigSchema: &plugin.ConnectionConfigSchema{
 			NewInstance: ConfigInstance,
 		},
+		RateLimiters: []*rate_limiter.Definition{
+			// FIXME: API Requests per 100 seconds: 5,000
+			// https://cloud.google.com/memorystore/docs/redis/quotas#per-second_api_requests_quota
+			{
```
Comment (misraved) [inline gcp/plugin.go:43]: @pdecat did you run into specific rate limit issues while testing out this table?

What would happen if we use the table without these rate limits?

Comment (pdecat) [reply gcp/plugin.go:43]: I actually only faced errors with the cluster API which has very low quotas. The retry code wouldn't suffice to handle those.

Also, removing the usage of matrix item function did help improve the situation as way less API calls are made.

Comment (misraved) [reply gcp/plugin.go:43]: @pdecat do we still need the `FIXME` comments? As per your test results, if you feel that they are the optimum values, I think we can have them in.

Comment (pdecat) [reply gcp/plugin.go:43]: Given the answers I got about this in Slack, especially the fact that [most plugins will back off and retry](most plugins will back off and retry), I do believe those are reasonable values.

I've removed the FIXMEs.

**Review by misraved — APPROVED** [review]

**Pull request merged**